src/transformers/commands/serving.py

```diff
@@ -98,131 +135,193 @@ def register_subcommand(parser: ArgumentParser):
         Args:
             parser: Root parser to register command-specific arguments
         """
-        serve_parser = parser.add_parser(
-            "serve", help="CLI tool to run inference requests through REST and GraphQL endpoints."
-        )
-        serve_parser.add_argument(
-            "--task",
-            type=str,
-            choices=get_supported_tasks(),
-            help="The task to run the pipeline on",
+        dataclass_types = (ServeArguments,)
+        serve_parser = parser.add_parser("serve", dataclass_types=dataclass_types)
+
+        group = serve_parser.add_argument_group("Positional arguments")
+        group.add_argument("model_name_or_path", type=str, default=None, help="Name of the pre-trained model.")
+        serve_parser.set_defaults(func=serve_command_factory)
+
+    def __init__(self, args: ServeArguments):
+        if not is_pydantic_available() or not is_fastapi_available() or not is_uvicorn_available():
+            raise ImportError("uvicorn, fastapi, and pydantic are required dependencies for the serving CLI.")
```

Comment:
consider adding a `transformers[serving]` extra for more centralized control over required deps ?

Reply:
There is already one! I'll edit that error so it references it, thank you